kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/manager/ResourceManager.java

```diff
@@ -118,9 +141,10 @@ private final <T extends HasMetadata> void createResource(boolean waitReady, T..
                 if (Objects.equals(resource.getKind(), KafkaTopic.RESOURCE_KIND)) {
                     continue;
                 }
+                assert type != null;
```

Comment:
You'd normally use Java Assertions to enforce a state invariant within an implementation.     Assertions have the advantage that you can disable the checks to benefit from extra speed.   You sometimes see assertions disabled in production.

I don't think your use-case here is a good one for Assertions.  I'd just throw an explicit exception.